libs/src/sdk/api/users/types.ts

```diff
@@ -1,5 +1,31 @@
 import { z } from 'zod'
+import type { CrossPlatformFile as File } from '../../types/File'
 import { HashId } from '../../types/HashId'
+import { isFileValid } from '../../utils/file'
+
+export const UpdateProfileSchema = z
+  .object({
+    userId: HashId,
+    profilePictureFile: z.optional(
+      z.custom<File>((data: unknown) => isFileValid(data as File))
+    ),
+    coverArtFile: z.optional(
+      z.custom<File>((data: unknown) => isFileValid(data as File))
+    ),
+    onProgress: z.optional(z.function().args(z.number())),
+    metadata: z
+      .object({
+        name: z.optional(z.string()),
+        bio: z.optional(z.string()),
+        location: z.optional(z.string()),
+        isDeactivated: z.optional(z.boolean()),
+        artistPickTrackId: z.optional(z.number())
+      })
+      .strict()
+  })
+  .strict()
+
+export type UpdateProfileRequest = z.input<typeof UpdateProfileSchema>
```

Comment:
This is super annoying but can you add a manual type declaration for `onProgress`? The inferred type doesn't have named arguments so it's not very useful for users. Here's how we did it for tracks https://github.com/AudiusProject/audius-protocol/blob/main/libs/src/sdk/api/tracks/types.ts#L114